src/test/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtilTest.java

```diff
@@ -67,8 +81,30 @@ public void testMetadataFilesGenerationAllFiles() throws Exception {
                 .sorted()
                 .collect(Collectors.toCollection(LinkedHashSet::new));
         checkstyleModules.removeAll(modulesContainingNoMetadataFile);
-        assertEquals("Number of generated metadata files dont match with number of checkstyle "
-                        + "module", checkstyleModules, metaFiles);
+
+        Assertions.assertEquals(checkstyleModules, metaFiles,
+                "Number of generated metadata files dont match with number of checkstyle module");
+
+        final String[] expectedErrorMessages = {
+            "41: " + getCheckMessage(MSG_DESC_MISSING, "AbstractJavadoc"),
+            "31: " + getCheckMessage(MSG_DESC_MISSING, "AbstractSuper"),
+            "29: " + getCheckMessage(MSG_DESC_MISSING, "AbstractParenPad"),
+            "44: " + getCheckMessage(MSG_DESC_MISSING, "AbstractClassCoupling"),
+            "43: " + getCheckMessage(MSG_DESC_MISSING, "AbstractHeader"),
+            "30: " + getCheckMessage(MSG_DESC_MISSING, "AbstractName"),
+            "26: " + getCheckMessage(MSG_DESC_MISSING, "AbstractAccessControlName"),
+        };
+
+        final String[] actualViolations = systemOut.getCapturedData().split("\\n");
+        final int javadocMetadataScraperLength = " [JavadocMetadataScraper]".length();
+        Arrays.setAll(actualViolations, id -> {
+            final String currString = actualViolations[id];
+            return currString.substring(currString.indexOf(':') + 1,
+                                        currString.length() - javadocMetadataScraperLength);
+        });
```

Comment:
This is a simpler approach, could be replaced with a regex-based approach if need raised, regex that can be used `(?<=.):.*(?=\\s\\[)`  and then extract the required group.